int64 action
---
int64 retval
